src/client/sky.cpp

```diff
@@ -640,113 +620,218 @@ void Sky::draw_sun(video::IVideoDriver *driver, float sunsize, const video::SCol
+				a.rotateVect(p3);
+				p.rotateXYBy(wicked_time_of_day * 360 - 90);
+				p.rotateXZBy(m_star_rotation);
+				p1.rotateXYBy(wicked_time_of_day * 360 - 90);
+				p1.rotateXZBy(m_star_rotation);
+				p2.rotateXYBy(wicked_time_of_day * 360 - 90);
+				p2.rotateXZBy(m_star_rotation);
+				p3.rotateXYBy(wicked_time_of_day * 360 - 90);
+				p3.rotateXZBy(m_star_rotation);
+				vertices[i * 4 + 0].Pos = p;
+				vertices[i * 4 + 0].Color = starcolor;
+				vertices[i * 4 + 1].Pos = p1;
+				vertices[i * 4 + 1].Color = starcolor;
+				vertices[i * 4 + 2].Pos = p2;
+				vertices[i * 4 + 2].Color = starcolor;
+				vertices[i * 4 + 3].Pos = p3;
+				vertices[i * 4 + 3].Color = starcolor;
+			}
+			driver->drawVertexPrimitiveList(vertices, m_star_count * 4,
+					indices, m_star_count, video::EVT_STANDARD,
+					scene::EPT_QUADS, video::EIT_16BIT);
+			delete[] indices;
+			delete[] vertices;
+#endif
+	}
 
-void Sky::draw_sky_body(std::array<video::S3DVertex, 4> &vertices, float pos_1, float pos_2, const video::SColor &c)
-{
-	/*
-	 * Create an array of vertices with the dimensions specified.
-	 * pos_1, pos_2: position of the body's vertices
-	 * c: color of the body
-	 */
+	void Sky::draw_sky_body(std::array<video::S3DVertex, 4> &vertices, float pos_1, float pos_2, const video::SColor &c)
+	{
+		/*
+		 * Create an array of vertices with the dimensions specified.
+		 * pos_1, pos_2: position of the body's vertices
+		 * c: color of the body
+		 */
 
-	const f32 t = 1.0f;
-	const f32 o = 0.0f;
-	vertices[0] = video::S3DVertex(pos_1, pos_1, -1, 0, 0, 1, c, t, t);
-	vertices[1] = video::S3DVertex(pos_2, pos_1, -1, 0, 0, 1, c, o, t);
-	vertices[2] = video::S3DVertex(pos_2, pos_2, -1, 0, 0, 1, c, o, o);
-	vertices[3] = video::S3DVertex(pos_1, pos_2, -1, 0, 0, 1, c, t, o);
-}
+		const f32 t = 1.0f;
+		const f32 o = 0.0f;
+		vertices[0] = video::S3DVertex(pos_1, pos_1, -1, 0, 0, 1, c, t, t);
+		vertices[1] = video::S3DVertex(pos_2, pos_1, -1, 0, 0, 1, c, o, t);
+		vertices[2] = video::S3DVertex(pos_2, pos_2, -1, 0, 0, 1, c, o, o);
+		vertices[3] = video::S3DVertex(pos_1, pos_2, -1, 0, 0, 1, c, t, o);
+	}
 
 
-void Sky::place_sky_body(
-	std::array<video::S3DVertex, 4> &vertices, float horizon_position, float day_position)
+	void Sky::place_sky_body(
+		std::array<video::S3DVertex, 4> &vertices, float horizon_position, float day_position)
 	/*
 	 * Place body in the sky.
 	 * vertices: The body as a rectangle of 4 vertices
 	 * horizon_position: turn the body around the Y axis
 	 * day_position: turn the body around the Z axis, to place it depending of the time of the day
 	 */
-{
-	for (video::S3DVertex &vertex : vertices) {
-		// Body is directed to -Z (south) by default
-		vertex.Pos.rotateXZBy(horizon_position);
-		vertex.Pos.rotateXYBy(day_position);
+	{
+		for (video::S3DVertex &vertex : vertices) {
+			// Body is directed to -Z (south) by default
+			vertex.Pos.rotateXZBy(horizon_position);
+			vertex.Pos.rotateXYBy(day_position);
+		}
```

Comment:
same here

Reply:
I cannot see any problem here.

Reply:
The whole function is indented with one more tab than necessary.